### PR TITLE
Integrate with Twint SDK

### PR DESCRIPTION
**Pull request by OscarSpruit**
## Description
- Adds support for the twint action
- The `TwintActionComponent` opens the Twint app and when it returns to the merchant's app it starts polling
- It uses a view that wraps a fragment, so we can handle to activity result returned by the twint app
    - This concept can be used by other SDK payment methods in the future

## Checklist 
- [x] PR is labelled 
- [x] Code is unit tested
- [x] Changes are tested manually

COAND-806

### CODE REVIEW

File: components-core/src/main/java/com/adyen/checkout/components/core/internal/util/StatusResponseUtils.kt [hunk 1]
```diff
@@ -18,8 +18,11 @@ object StatusResponseUtils {
...
     const val RESULT_CANCELED = "canceled"
 
-    @JvmStatic
     fun isFinalResult(statusResponse: StatusResponse): Boolean {
         return RESULT_PENDING != statusResponse.resultCode
     }
+
+    fun isPendingResult(statusResponse: StatusResponse): Boolean {
+        return RESULT_PENDING == statusResponse.resultCode
+    }
```
[the remark this answers is not in the document]
Comment (OscarSpruit) [reply components-core/src/main/java/com/adyen/checkout/components/core/internal/util/StatusResponseUtils.kt:27]: Good catch! I'll remove it

File: econtext/build.gradle [hunk 2]
```diff
@@ -48,6 +48,7 @@ dependencies {
     //Tests
     testImplementation project(':3ds2')
     testImplementation project(':test-core')
+    testImplementation project(':twint')
```
[the remark this answers is not in the document]
Comment (OscarSpruit) [reply econtext/build.gradle:51]: Yes for some reason it doesn't compile without

File: instant/src/main/java/com/adyen/checkout/instant/ActionHandlingMethod.kt [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+enum class ActionHandlingMethod {
+    /**
+     * The action will be handled in a native way (e.g. using a SDK). **If** there is no way to handle the action
+     * natively, then a fallback method will be used (e.g. a web flow).
+     */
+    PREFER_NATIVE,
+
+    /**
+     * The action will be handled with a web flow. **If** there is no way to handle the action with a web flow, then
+     * other methods might be used.
```
[the remark this answers is not in the document]
Comment (OscarSpruit) [reply instant/src/main/java/com/adyen/checkout/instant/ActionHandlingMethod.kt:23]: I think `PREFER_WEB` is more explicit. We will try to handle the action in a web flow and if not possible we might try with a native solution

File: instant/src/main/java/com/adyen/checkout/instant/InstantPaymentConfiguration.kt [hunk 4]
```diff
@@ -25,12 +25,15 @@ import java.util.Locale
...
-class InstantPaymentConfiguration private constructor(
+class InstantPaymentConfiguration
+@Suppress("LongParameterList")
+private constructor(
     override val shopperLocale: Locale?,
     override val environment: Environment,
     override val clientKey: String,
     override val analyticsConfiguration: AnalyticsConfiguration?,
     override val amount: Amount?,
+    val actionHandlingMethod: ActionHandlingMethod?,
```
[the remark this answers is not in the document]
Comment (OscarSpruit) [reply instant/src/main/java/com/adyen/checkout/instant/InstantPaymentConfiguration.kt:36]: It's similar as the fields in `CardConfiguration`, so I think it should be public

File: twint/src/main/java/com/adyen/checkout/twint/internal/ui/DefaultTwintDelegate.kt [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+        if (action.sdkData == null || sdkData !is TwintSdkData) {
+            emitError(ComponentException("SDK Data is null or of wrong type"))
+            return
+        }
+
+        @Suppress("UNCHECKED_CAST")
+        this.action = action as SdkAction<TwintSdkData>
+
+        initState(action)
+        launchAction(sdkData)
```
[the remark this answers is not in the document]
Comment (OscarSpruit) [reply twint/src/main/java/com/adyen/checkout/twint/internal/ui/DefaultTwintDelegate.kt:127]: When restoring the state `handleAction` is not called. The `initState` function does check this boolean. 
And when `handleAction` is called twice by accident, then it stops and starts polling again. So I think it's fine

File: ui-core/src/main/java/com/adyen/checkout/ui/core/AdyenComponentView.kt [hunk 6]
```diff
@@ -108,6 +111,7 @@ class AdyenComponentView @JvmOverloads constructor(
                     coroutineScope = lifecycleOwner.lifecycleScope,
                 )
             }
+            .flowWithLifecycle(lifecycleOwner.lifecycle)
```
[the remark this answers is not in the document]
Comment (OscarSpruit) [reply ui-core/src/main/java/com/adyen/checkout/ui/core/AdyenComponentView.kt:114]: Surprisingly not 🤔

File: econtext/build.gradle (hunk 2 above)
[the remark this answers is not in the document]
Comment (araratthehero) [reply econtext/build.gradle:51]: For me it does compile. What error does it show for you?

File: instant/src/main/java/com/adyen/checkout/instant/ActionHandlingMethod.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (araratthehero) [reply instant/src/main/java/com/adyen/checkout/instant/ActionHandlingMethod.kt:23]: I see, does it then make sense to adjust the comment?

```suggestion
     * native method will be used.
```
     * other methods might be used.`

File: twint/src/main/java/com/adyen/checkout/twint/internal/ui/DefaultTwintDelegate.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (araratthehero) [reply twint/src/main/java/com/adyen/checkout/twint/internal/ui/DefaultTwintDelegate.kt:124]: I see, makes sense.

File: econtext/build.gradle (hunk 2 above)
[the remark this answers is not in the document]
Comment (araratthehero) [reply econtext/build.gradle:51]: I see, probably that is why we have the rest of the project testImplementations too.

**Review by araratthehero — APPROVED** [review]

**Pull request merged**